lib/heartcheck/checks/newrelic.rb

```diff
@@ -0,0 +1,18 @@
+module Heartcheck
+  module Checks
+    class Newrelic < Base
+      def validate
+        NewRelic::Agent.notify_error('Heartcheck monitoring', :expected)
```

Comment:
I think doing this way, it will show a lot of errors in new-relic panel, isn't it?

Maybe you could use the method [increment_metric](http://www.rubydoc.info/github/newrelic/rpm/master/NewRelic/Agent#increment_metric-instance_method) to test the connectivity, wdyt? (And this could be used as a metric to show how many monitoring run in a period)

```ruby
NewRelic::Agent.notify_error
=> NoMethodError: undefined method `notify_error' for NewRelic::Agent:Module

NewRelic::Agent.notice_error
=> ArgumentError: wrong number of arguments (given 0, expected 1..2)
```

Other thing...

Where did you find the method `notify_error` ? I could only find the method [notice_error](http://www.rubydoc.info/github/newrelic/rpm/master/NewRelic/Agent#notice_error-instance_method)

Reply:
> I think doing this way, it will show a lot of errors in new-relic panel, isn't it?
I guess no, because of the `:expected`.

> Where did you find the method notify_error ?
Probably a typo 😆

Reply:
Yes it was a typo.

Reply:
@fabioperrella, since we are using the `:expected` option the behavior we are expecting, is:

```
In the Java and Ruby agents, you can also mark an error as "expected." 
An expected error is still reported to the collector and visible in New Relic APM UI and New 
Relic Insights, but it doesn't affect the Apdex or error rate (including alert 
conditions based on error rate)
```
https://docs.newrelic.com/docs/agents/manage-apm-agents/agent-data/manage-errors-apm-collect-ignore-or-mark-expected#expected-links

So it will not affect the metrics at all.

And I was looking for the method you mentioned(increment_metric), and I believe that the `notice_error` makes more sence for this case. What do you think?

Reply:
Even if it will not affect the Apdex, I think it is better to increment a metric instead of report a non existing error. And reporting the metric, it is possible to test the connection, isn't it?

I would use the method `notice_error` only if it was the only way.

Reply:
Done! @fabioperrella